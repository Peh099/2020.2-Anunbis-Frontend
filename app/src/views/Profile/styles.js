import styled from 'styled-components';
import Button from '../../components/Button';

export const Conteiner = styled.div`
  display: ${(props) => (props.display ? props.display : '')};
  flex-direction: column;
  align-items: center;
  width: ${(props) => (props.width ? props.width : 'min(65vw, 560px)')};
  height: ${(props) => (props.height ? props.height : 'min(65vh, 560px)')};
  background-color: ${(props) => (props.backColor ? props.backColor : '')};
  border-radius: 20px;
  text-align: ${(props) => (props.txtAlign ? props.txtAlign : '')};

  p {
    font-size: 12px;
    margin-left: 5px;
    margin-top: 14px;
    font-family: 'Comfortaa', cursive;
    flex-wrap: wrap;
  }

  @media (max-height: 600px) {
    overflow-y: auto;
  }
`;

export const BtnReset = styled(Button)`
  margin-top: 20px;
  margin-bottom: 20px;
  @media (max-height: 660px) {
    font-size: 10px;
  }
`;

export const BtnExcluir = styled(Button)`
  color: white;
  margin-top: 20px;
  margin-bottom: 20px;
  border: none;

  &:hover,
  &focus {
    color: var(--black);
  }
  @media (max-height: 660px) {
    font-size: 10px;
  }
`;

export const Title = styled.h4`
  display: flex;
  justify-content: center;
  margin-bottom: 20px;
  margin-top: 40px;
  @media (max-height: 660px) {
    font-size: 14px;
  }
`;