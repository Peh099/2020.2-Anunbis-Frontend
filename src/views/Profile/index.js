import React from 'react';
import styled from 'styled-components';
import Button from '../../components/Button';
import ResetPassword from '../../components/ResetPassword';
import {logOut} from '../../services/authentication'
import {useHistory} from "react-router-dom";

const Container = styled.div`
    display: ${props => props.display?props.display:''};
    flex-direction:column;
    align-items: center;
    width: ${props => props.width?props.width:'460px'};
    height: ${props => props.height?props.height:'460px'};
    background-color: ${props => props.backColor?props.backColor:''};
    border-radius: 20px;
    text-align: ${props => props.txtAlign?props.txtAlign:''};
    /* justify-content: center; */

    p{
        font-size: 14px;
        margin-left: 5px;
    }

`;

const ContentExclude = styled.div`
    position: absolute;
    width: 200px;
    height: 100px;
    display: flex;
    flex-direction: column;
    background-color: #FFF9C4;
    margin-top: 180px;
    border-radius: 10px;

    header{
        border-top-left-radius: 10px;
        border-top-right-radius: 10px;
        background-color: #212121;
        color: white;
        text-align: center;
    }

`

const BtsExclude = styled.div`
    display: flex;
    flex-direction: row;
    background-color: inherit;
    height: inherit;
    justify-content: center;
    align-items: center;
    border-radius: 10px;

    Button{
        color: white;
    }

`

const BtnReset = styled(Button)`
    margin-top: 20px;
    margin-bottom: 20px;
`

const BtnExcluir = styled(Button)`
    color: white;
    margin-top: 20px;
    margin-bottom: 20px;
`

const Title = styled.h4`
  display: flex;
  justify-content: center;
  /* align-items: center; */
  margin-bottom:20px;
  margin-top: 40px;
`;

export default function Profile(){

    const student = {
        name: JSON.parse(localStorage.getItem('student')).name,
        email: JSON.parse(localStorage.getItem('student')).email,
        course: JSON.parse(localStorage.getItem('student')).id_course
    }

    const [resetPassword, setResetPassword] = React.useState("");

    function makeResetPassword() {
        return(
            setResetPassword(<ResetPassword onClick={() => setResetPassword("")} student={student} />)
        )
    }

    const [excludeAccount, setExcludeAccount] = React.useState("");
    const history = useHistory();

    function makeExcludeAccount(){
        return(
            setExcludeAccount(
            <ContentExclude >
                <header>
                Tem certeza?
                </header>
                
                <BtsExclude>
                    <Button backColor='#26A69A' padding='10px 7px' text='SIM' onClick={() => {logOut(); history.push('/user/login')}}/>
                    <Button backColor='#26A69A' padding='10px 7px' text='NÃO' onClick={() => setExcludeAccount('')}/>
                </BtsExclude>
            </ContentExclude>)
        )
    }

    return (
    <Container display='flex' backColor='#E0E0E0'>
    {resetPassword}
    {excludeAccount}

    <Title>
        Configurações de conta
    </Title>
    
    <Container backColor='#FFFFFF' width='240px' height='115px'>
        <p>Nome Completo: {student.name}</p>
        <p>E-mail: {student.email}</p>
        <p>Curso: {student.course}</p>
    </Container>

    {console.log(localStorage.getItem("student"))}

    <BtnReset text='ALTERAR SENHA' backColor='#FFF9C4' padding='5px' onClick={()=> makeResetPassword()}/>

    <Container txtAlign='center' backColor='#FFFFFF' width='430px' height='115px'>
        <p>Quantidade de avaliações realizadas: </p>
        <p>Quantidade de pessoas que concordaram com suas avaliações: </p>
        <p>Quantidade de pessoas que discordaram com suas avaliações: </p>
    </Container>

    <BtnExcluir text='EXCLUIR CONTA' backColor='#F44336' padding='5px' onClick={()=> makeExcludeAccount()}/>

    </Container>)
}