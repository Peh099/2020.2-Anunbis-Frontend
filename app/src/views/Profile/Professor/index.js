import React, { useEffect } from 'react';
import { Conteiner } from './styles';
import Users from '../../../services/Users';
import { getPosts } from '../../../services/Posts';

const Header = () => {
  const professor = Users.whoAuthenticated().data();
  return (
    <Conteiner backColor="#FFFFFF" width="240px" height="115px">
      <p>Nome Completo: {professor.name}</p>
      <p>E-mail: {professor.email}</p>
    </Conteiner>
  );
};

const Body = () => {
  const [posts, setPosts] = React.useState([]);

  useEffect(() => {
    getPosts(setPosts);
  }, []);

  function getRating() {
    return (
      posts.reduce((accumulator, p) => accumulator + p.rating, 0) / posts.length
    );
  }

  console.log(posts);
  return (
    <Conteiner
      txtAlign="center"
      backColor="#FFFFFF"
      width="430px"
      height="115px"
    >
      <p>Quantidade de avaliações recebidas: {posts.length}</p>
      <p>Pontuação média recebida: {getRating()}</p>
    </Conteiner>
  );
};
function ProfileProfessor({ children }) {
  return children;
}

ProfileProfessor.Header = Header;
ProfileProfessor.Body = Body;

export default ProfileProfessor;
