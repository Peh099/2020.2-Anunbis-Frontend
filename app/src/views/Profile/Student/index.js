import React, { useEffect } from 'react';
import Users from '../../../services/Users';
import { Conteiner } from './styles';
import { getCourses } from '../../../services/Courses';
import api from '../../../services/Api';

const Header = () => {
  const [courses, setCourses] = React.useState([]);
  const student = Users.whoAuthenticated().data();

  useEffect(() => {
    getCourses(setCourses);
  }, []);

  function getCourseName() {
    let cont = 0;
    let name;
    while (cont < courses.length) {
      if (courses[cont].id_course === student.id_course) {
        name = courses[cont].name;
      }
      cont += 1;
    }
    return name;
  }
  return (
    <Conteiner backColor="#FFFFFF" width="240px" height="115px">
      <p>
        Nome Completo:
        {student.name}
      </p>
      <p>
        E-mail:
        {student.email}
      </p>
      <p>
        Curso:
        {getCourseName()}
      </p>
    </Conteiner>
  );
};

const Body = () => {
  const [posts, setPosts] = React.useState([]);

  useEffect(() => {
    api.get('post').then((response) => {
      if (response.status === 200) setPosts(response.data);
    });
  }, []);
  console.log(posts);

  function countAgree() {
    return posts.reduce(
      (accumulator, p) => accumulator + p.feedbacks.agrees,
      0,
    );
  }

  function countDisagree() {
    return posts.reduce(
      (accumulator, c) => accumulator + c.feedbacks.disagrees,
      0,
    );
  }

  return (
    <Conteiner
      txtAlign="center"
      backColor="#FFFFFF"
      width="430px"
      height="115px"
    >
      <p>Quantidade de avaliações realizadas: {posts.length}</p>
      <p>
        Quantidade de pessoas que concordaram com suas avaliações:{' '}
        {countAgree()}
      </p>
      <p>
        Quantidade de pessoas que discordaram com suas avaliações:{' '}
        {countDisagree()}
      </p>
    </Conteiner>
  );
};
function ProfileStudent({ children }) {
  return children;
}

ProfileStudent.Header = Header;
ProfileStudent.Body = Body;

export default ProfileStudent;
