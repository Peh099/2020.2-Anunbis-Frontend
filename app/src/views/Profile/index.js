import React from 'react';
import ResetPassword from '../../components/ResetPassword';
import { Conteiner, BtnReset, BtnExcluir, Title } from './styles';
import ProfileStudent from './Student';
import ProfileProfessor from './Professor';
import ExcludeAccount from '../../components/ExcludeAccount';
import Users from '../../services/Users';

export default function Profile() {
  const [excludeAcc, setExcludeAcc] = React.useState('');
  const [resetPassword, setResetPassword] = React.useState('');
 
  const ProfileSpecific =
    Users.whoAuthenticated() === Users.STUDENT
      ? ProfileStudent
      : ProfileProfessor;

  function makeReset() {
    setResetPassword(<ResetPassword onClick={() => setResetPassword('')} />);
  }

  function makeExclude() {
    setExcludeAcc(<ExcludeAccount close={() => setExcludeAcc('')} />);
  }

  return (
    <Conteiner display="flex" backColor="#E0E0E0">
      {resetPassword} 
      {excludeAcc}
      <Title>Configurações de conta</Title>
      <ProfileSpecific.Header />
      <BtnReset
        text="ALTERAR SENHA"
        backColor="#FFF9C4"
        padding="5px"
        onClick={makeReset}
      />
      <ProfileSpecific.Body />
      <BtnExcluir
        text="EXCLUIR CONTA"
        backColor="#F44336"
        padding="5px"
        onClick={makeExclude}
      />
    </Conteiner>
  );
}
