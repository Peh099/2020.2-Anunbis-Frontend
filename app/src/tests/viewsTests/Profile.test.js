import renderer from 'react-test-renderer';
import Profile from '../../views/Profile/index';

describe('Snapshot Profile view', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Profile />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
