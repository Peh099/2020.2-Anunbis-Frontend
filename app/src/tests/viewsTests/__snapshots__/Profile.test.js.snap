// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Profile view matches the snapshot 1`] = `
<div
  className="sc-dIsUp joMOBp"
  display="flex"
>
  
  
  <h4
    className="sc-hBMUJo CSbtl"
  >
    Configurações de conta
  </h4>
  <div
    className="sc-fFSPTT iRkqee"
    height="115px"
    width="240px"
  >
    <p>
      Nome Completo: 
    </p>
    <p>
      E-mail: 
    </p>
  </div>
  <button
    className="sc-jSFjdj cjmoSN sc-bqGGPW gxWuCk"
    onClick={[Function]}
  >
    ALTERAR SENHA
  </button>
  <div
    className="sc-fFSPTT dqAmUg"
    height="115px"
    width="430px"
  >
    <p>
      Quantidade de avaliações recebidas: 
      0
    </p>
    <p>
      Pontuação média recebida: 
      0
    </p>
  </div>
  <button
    className="sc-jSFjdj kZEVrR sc-ksluID dbTocj"
    onClick={[Function]}
  >
    EXCLUIR CONTA
  </button>
</div>
`;
